### DISCUSSION TITLE
Unable to wrap the upload element in rich text with link

### DISCUSSION THREAD

**Discussion by EasonSoong** [payloadcms/payload#1231] [Q&A] [answered]
# Bug Report

Use link to wrap upload element don't work.
![image](https://user-images.githubusercontent.com/57319837/194699033-d9078ea6-a0a2-400e-8996-44681f8bff14.png)

## Current Behavior

Enter the link form and submit, but nothing happen.

## Expected Behavior

Should be able to wrap an upload element with link to make it clickable.

## Steps to Reproduce

1. Insert an upload element in rich text.
2. Click the upload element and click the link to wrap it.
3. Enter the link form and confirm.  ![image](https://user-images.githubusercontent.com/57319837/194699275-7fa447e1-1784-4024-8f62-80ead96c0e83.png)
4. Nothing happen.

## Detailed Description

payload@1.1.7

**Comment by jmikrut** [comment, marked as answer]
Hey @EasonSoong — this is actually intended. If you want to link your uploads, you can add custom upload fields like seen in the demo, in the Page -> Layout -> Content rich text fields:

<img width="1722" alt="Screen Shot 2022-10-09 at 8 33 08 PM" src="https://user-images.githubusercontent.com/6651896/194786599-2ee08732-e18b-4738-95df-77859a569cdb.png">

Here are some docs about this:
https://payloadcms.com/docs/fields/rich-text#upload-element

It might be possible in the future to extend the Link component, but it's tough because of the way that SlateJS works. It would be an "element within an element" as far as Slate is concerned. Not gonna say it's impossible, but the UX of it all could be questionable as well. So we prefer the above method to link uploads within rich text.

Gonna convert this to a discussion for now but would love to know what you think!
